src/plots/cartesian/layout_defaults.js

```diff
@@ -145,6 +145,7 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
             coerce('spikethickness');
             coerce('spikedash');
             coerce('spikemode');
+            coerce('spikesnap');
```

Comment:
If you're interested, we should covert this `coerce` block to a `coerce2` pattern (see [example](https://github.com/plotly/plotly.js/blob/ef181daa3f18483af769cb776059b9709b17311b/src/plots/cartesian/axis_defaults.js#L88-L95)) where `showspikes: true` is implied whenever one of the `spike*` is correctly set by the user. Non-blocking to get this PR merged though.

Reply:
Thanks! Done.

Reply:
Wow. Thanks for doing this!